src/lucky/cookies/cookie_jar.cr

```diff
@@ -1,72 +1,113 @@
+  private def initialize(@cookies : HTTP::Cookies)
   end
 
-  def get(key : Key) : HTTP::Cookie
-    store[key.to_s]
+  def raw : HTTP::Cookies
+    cookies
   end
 
-  def get?(key : Key) : Lucky::MaybeCookie
-    store[key.to_s]? || Lucky::NullCookie.new
+  def clear : Void
+    cookies.each do |cookie|
+      delete cookie.name
+    end
   end
 
-  def set(key : Key, value : String) : HTTP::Cookie
-    @changed = true
-    store[key.to_s] = HTTP::Cookie.new(
-      key.to_s,
-      value,
-      expires: settings.default_expiration.from_now,
-      http_only: true,
-    )
+  def delete(key : Key) : Void
+    raw[key.to_s].try &.expires(1.year.ago).value("")
+  end
+
+  def get_raw(key : Key) : HTTP::Cookie
+    get_raw?(key) || raise "No cookie for '#{key}'"
   end
 
-  def expire(key : Key)
-    @changed = true
-    store[key.to_s].expires(1.second.ago)
+  def get_raw?(key : Key) : HTTP::Cookie?
+    cookies[key.to_s]?
   end
 
-  def changed?
-    @changed
+  def get(key : Key) : String
+    get?(key) || raise "No cookie for '#{key}'"
   end
 
-  def each(&block : HTTP::Cookie ->)
-    @store.each do |cookie|
-      yield cookie
+  def get?(key : Key) : String?
+    cookies[key.to_s]?.try do |cookie|
+      decrypt(cookie.value, cookie.name)
     end
+  rescue OpenSSL::Cipher::Error
+    nil
   end
 
-  def clear
-    @store = HTTP::Cookies.new
+  def set(key : Key, value : String) : HTTP::Cookie
+    set_raw key, encrypt(value)
+  end
+
+  def set_raw(key : Key, value : String) : HTTP::Cookie
+    cookies[key.to_s] = HTTP::Cookie.new(
+      name: key.to_s,
+      value: value,
+      http_only: true,
+    ).tap do |cookie|
+      settings.on_set.try(&.call(cookie))
+    end
   end
 
-  def session_cookie
-    if get?(Lucky::SessionCookie.settings.key).is_a?(Lucky::NullCookie)
-      HTTP::Cookie.new(Lucky::SessionCookie.settings.key, "{}")
+  private def encrypt(raw_value : String) : String
+    encrypted = encryptor.encrypt(raw_value)
+
+    String.build do |value|
+      value << ENCRYPTION_PREFIX
+      value << Base64.strict_encode(encrypted)
+    end
+  end
+
+  private def decrypt(cookie_value : String, cookie_name : String) : String
+    if encrypted?(cookie_value)
+      base_64_encrypted_part = cookie_value.lchop(ENCRYPTION_PREFIX)
+      decoded = Base64.decode(base_64_encrypted_part)
+      String.new(encryptor.decrypt(decoded))
     else
-      get(Lucky::SessionCookie.settings.key)
+      raise <<-ERROR
+      It looks like this cookie's value is not encrypted by Lucky. This likely means the cookie was set by something other than Lucky, like a client side script.
+
+      You can access the raw value by using 'get_raw':
+
+          cookies.get_raw(:#{cookie_name}).value
+
+      ERROR
     end
   end
 
-  def who_took_the_cookies_from_the_cookie_jar?
-    raise "Edward Loveall"
+  private def encrypted?(value : String) : Bool
+    value.starts_with?(ENCRYPTION_PREFIX)
```

Comment:
This isn't _quite_ true since it could have been encrypted elsewhere, but `encrypted_with_lucky?` is too long of a name. What do you think? Doesn't matter?

Reply:
I like that! I think it adds clarity without being too long